modules/sfp_seon.py

```diff
@@ -0,0 +1,256 @@
+            "PROVIDER_TELCO",
+            "PHONE_NUMBER_TYPE",
+            "WEBSERVER_TECHNOLOGY",
+            "RAW_RIR_DATA"
+        ]
+
+    def query(self, qry, eventName):
+        if eventName == "IP_ADDRESS" or eventName == "IPV6_ADDRESS":
+            queryString = f"https://api.seon.io/SeonRestService/ip-api/v1.0/{qry}"
+        elif eventName == "EMAILADDR":
+            queryString = f"https://api.seon.io/SeonRestService/email-api/v2.0/{qry}"
+        elif eventName == "PHONE_NUMBER":
+            queryString = f"https://api.seon.io/SeonRestService/phone-api/v1.0/{qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'X-API-KEY': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        return json.loads(res['content'])
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts['api_key'] == "":
+            self.sf.error("You enabled sfp_seon but did not set an API key!")
+            self.errorState = True
+            return
+
+        if self.errorState:
+            return None
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug(f"Skipping {eventData}, already checked.")
+            return None
+        else:
+            self.results[eventData] = True
+
+        if eventName == "IP_ADDRESS" or eventName == "IPV6_ADDRESS":
+            data = self.query(eventData, eventName)
+            if data is None:
+                return None
+
+            resultSet = data.get('data')
+            if resultSet:
+                if resultSet.get('score') >= self.opts['fraud_threshold']:
+                    maliciousDesc = f"SEON [{eventData}]\n - FRAUD SCORE: {resultSet.get('score')}"
+                    evt = SpiderFootEvent("MALICIOUS_IPADDR", maliciousDesc, self.__name__, event)
+                    self.notifyListeners(evt)
+
+                if resultSet.get('country'):
+                    location = ', '.join(filter(None, [resultSet.get('city'), resultSet.get('state_prov'), resultSet.get('country')]))
+                    location += f"\n-Latitude: {resultSet.get('latitude')}\n-Longitude: {resultSet.get('longitude')}"
+                    evt = SpiderFootEvent('GEOINFO', location, self.__name__, event)
+                    self.notifyListeners(evt)
+
+                if resultSet.get('open_ports'):
+                    for port in resultSet.get('open_ports'):
+                        evt = SpiderFootEvent('TCP_PORT_OPEN', f"{eventData}:{port}", self.__name__, event)
+                        self.notifyListeners(evt)
+
+                if resultSet.get('tor'):
```

Comment:
These following events should only be reported if they are positive. i.e. I don't think it's valuable to report that something is NOT a VPN, for example, since most things won't be.